testament/specs.nim

```diff
@@ -350,42 +351,43 @@ proc parseSpec*(filename: string): TSpec =
           # Valgrind only supports OSX <= 17.x
           result.useValgrind = disabled
       of "disabled":
-        case e.value.normalize
-        of "y", "yes", "true", "1", "on": result.err = reDisabled
-        of "n", "no", "false", "0", "off": discard
-        of "win", "windows":
-          when defined(windows): result.err = reDisabled
-        of "linux":
-          when defined(linux): result.err = reDisabled
-        of "bsd":
-          when defined(bsd): result.err = reDisabled
-        of "osx", "macosx": # xxx remove `macosx` alias?
-          when defined(osx): result.err = reDisabled
-        of "unix":
-          when defined(unix): result.err = reDisabled
-        of "posix":
-          when defined(posix): result.err = reDisabled
-        of "travis": # deprecated
-          if isTravis: result.err = reDisabled
-        of "appveyor": # deprecated
-          if isAppVeyor: result.err = reDisabled
-        of "azure":
-          if isAzure: result.err = reDisabled
-        of "32bit":
-          if sizeof(int) == 4:
-            result.err = reDisabled
-        of "freebsd":
-          when defined(freebsd): result.err = reDisabled
-        of "arm64":
-          when defined(arm64): result.err = reDisabled
-        of "i386":
-          when defined(i386): result.err = reDisabled
-        of "openbsd":
-          when defined(openbsd): result.err = reDisabled
-        of "netbsd":
-          when defined(netbsd): result.err = reDisabled
-        else:
-          result.parseErrors.addLine "cannot interpret as a bool: ", e.value
+        for v in e.value.normalize.split(";"):
+          case v
+          of "y", "yes", "true", "1", "on": result.err = reDisabled
+          of "n", "no", "false", "0", "off": discard
```

Comment:
How should these be fixed? i'm inclined to add a "disabled" entry for each platform.

Reply:
disabled: One per platform makes sense, this states that the test is unsupported on that platform and never will be. So boolean value don't belong anymore. Semicolon separators aren't bad, but a change that we didn't think through and we've got a much bigger overhaul to do down the road. So the existing multiple occurrences works.

known issue: one per reason (usually link) and marks the whole test as not expected to work. And can combine all the mentioned reasons as one big ol' reason on output.

This isn't a perfect design, but the parseCfg parser has limits, and it's workable for now. What we haven't explored with parseCfg is whether we can use sections inside testament. That's ok to come back to afterwards because that entails a more significant overhaul of how we declare tests.

Reply:
> How should these be fixed? i'm inclined to add a "disabled" entry for each platform.

About half the tests are disabled, for those I would take the following approach:
1. If it's specific to a platform, then nothing to do here
2. If it's true, then note the reason out of git history, eg: I disabled a few because of nkError changes, so these are known issues
3. otherwise if true, then delete because they're silly or flag for discussion by using the known issue field "discuss before merging" 😁

Reply:
This is now out of date, see conversation below.